### ISSUE TITLE
Cannot access property/attribute of returned record

### ISSUE THREAD

**Issue by gvanto** [baopham/laravel-dynamodb#277]
**Describe the bug**

Cannot access property 'target' of returned record.

TestController:

`class TestController
{
    public function index(Request $request)
    {
        /** @var DynamoDBRedirect $redir */
        $redir = DynamoDBRedirect::where(['source' => 'foo'])->first();

        dd($redir->target);

        // Code used to create the record (OK)
//        $model = new DynamoDBRedirect([
//            'source' => 'foo',
//            'target' => 'bar',
//        ]);
//
//        $model->save();
    }
}`

DynamoDBRedirect.php:

`<?php

namespace App\Models;

use BaoPham\DynamoDb\DynamoDbModel;

class DynamoDBRedirect extends DynamoDbModel
{
    public $table = 'redirects';

    public $primaryKey = 'source';

    public $source;

    public $target;

    protected $fillable = [
        'source',
        'target'
    ];

    protected $guarded = [];
}`

Result on test controller output:

![image](https://github.com/baopham/laravel-dynamodb/assets/8372918/eb19bc04-9247-4027-a2da-d1af966bc3b4)

Result with dd($redir):

![image](https://github.com/baopham/laravel-dynamodb/assets/8372918/4468dec4-9e7b-4c2b-b96e-66f6eb44007c)

**Schema**
table=redirects

Item exists in DDB, call from CLI:

`aws dynamodb get-item --consistent-read \
> --table-name redirects \
> --key '{ "source": {"S": "foo"}}' \
> --endpoint-url=http://localhost:8000
{
    "Item": {
        "created_at": {
            "S": "2024-01-09T14:23:01+00:00"
        },
        "source": {
            "S": "foo"
        },
        "updated_at": {
            "S": "2024-01-09T14:23:01+00:00"
        },
        "target": {
            "S": "bar"
        }
    }
}`

Describe your table schema:
* Primary key / composite key
* Any index?

**Debug info**

**Version info**

* Laravel: 10.x
* laravel-dynamodb: latest

**Comment by gvanto** [comment]
Maybe I am not "finding" the record correctly?

Why does it use $model->find() rather than MyModel::where() ?

I can't see a find() method anywhere in DynamoDbModel? So if my class extends that class, how does $model->find() work?

Assuming that MyModel extends DynamoDbModel.

Finding the docs quite unclear on this ... (sample of [good docs](https://github.com/kitar/laravel-dynamodb?tab=readme-ov-file#extending-the-base-model)).

![image](https://github.com/baopham/laravel-dynamodb/assets/8372918/ba6bab31-92ae-460f-925f-150e2c9c437b)

**Comment by nelson6e65** [comment]
Yes, you can use the static `find()` normally in your model. `DynamoDbModel` inherits from `Model` all methods, just adjusting the required ones to work with Dynamo DB.

@gvanto You have some problems in your model definition:

1. You **MUST NOT** add your database attributes as model properties. That will prevent to hydrated as attributes when fetched from the database; remove them.
2. You are using `$fillable` and `$guarded` together. It is not recommended to use `$fillable` and `$guarded` together because they have opposite effects; `$fillable` allows only the specified attributes to be mass assigned, while `$guarded` blocks only the specified attributes from being mass assigned. 


> As recommendation, you can use a sub-namespace for DynamoDb models (`App\Models\Dynamo\...`) instead of using a prefix in the model name. With this, you can take advantage of the automagic benefits of Laravel's models for table names and else.

> And, sorry for the lack in the documentation. I had not have much time to make some improvements I planned to. That's in the roadmap tho.
